app/controllers/products_controller.rb

```diff
@@ -0,0 +1,111 @@
+class ProductsController < ApplicationController
+  before_action :current_user
+  before_action :require_login, only: [:new, :create, :edit, :update, :retire]
+  before_action :check_user_product, only: [:review]
+  before_action :check_user_id, only: [:retire, :edit, :update]
+
+  def index
+    @categories = Category.all
+    @merchants = User.all
+    @products = Product.order(:created_at).where(retired: false)
+    case params[:type]
+
+      when "merch"
+        @products = @products.where("user_id = #{params[:order]}")
+      when "cat"
+        prod = Product.where(retired: false)
+        @products = prod.select {|product|
```

Comment:
What is this section of logic doing? Comments are helpful when using shortcut methods like select.